plugins/apps/threatbus_cif3/plugin.py

```diff
@@ -0,0 +1,98 @@
+from queue import Queue
+import threading
+from cifsdk.client.http import HTTP as Client
+
+from threatbus_cif3.message_mapping import map_to_cif
+import threatbus
+
+"""Threatbus - Open Source Threat Intelligence Platform - plugin for CIFv3"""
+
+plugin_name = "cif3"
+cif = None
+lock = threading.Lock()
+
+
+def validate_config(config):
+    assert config, "config must not be None"
+    config["tags"].get(list)
+    config["tlp"].get(str)
+    config["confidence"].as_number()
+    config["group"].get(str)
+    if config["api"].get(dict):
+        config["api"]["host"].get(str)
+        config["api"]["ssl"].get(bool)
+        config["api"]["token"].get(str)
+
+
+def receive_intel_from_backbone(watched_queue, config):
+    """Reports / publishes intel items back to the given CIF endpoint.
+        @param watched_queue The py queue from which to read messages to submit on to CIF
+    """
+    global logger, cif, lock
+    logger.debug(
+        f"Entering receive_intel_from_backbone func to monitor queue {watched_queue}..."
+    )
+    if not cif:
+        logger.debug("No global CIF found. Exiting function")
+        return
+    while True:
+        intel = watched_queue.get()
+        if not intel:
+            logger.debug("Sighting msg was empty")
+            continue
+        cif_mapped_intel = map_to_cif(intel, logger, config)
```

Comment:
This should not pass the `config` directly to the mapping function. The mapping should be independent of what configuration provider we use (currently `confuse`). Instead, pass the required arguments directly (`confidence`, `tags`, `tlp`, and `group`).

Reply:
Perhaps something like this in the receive_intel_from_backbone func
```suggestion
    confidence = config["confidence"].as_number()
    if not confidence:
        confidence = 5

    tags = config["tags"].get(list)
    tlp = config["tlp"].get(str)
    group = config["group"].get(str)
```

```suggestion
        cif_mapped_intel = map_to_cif(intel, logger, confidence, tags, tlp, group)
```